pkg/volume/csi/csi_plugin.go

```diff
@@ -160,19 +160,30 @@ func (h *RegistrationHandler) DeRegisterPlugin(pluginName string) {
 func (p *csiPlugin) Init(host volume.VolumeHost) error {
 	p.host = host
 
-	// Initializing csiDrivers map and label management channels
-	csiDrivers = csiDriversStore{driversMap: map[string]csiDriver{}}
-	nim = nodeinfomanager.NewNodeInfoManager(host.GetNodeName(), host.GetKubeClient(), host.GetCSIClient())
+	kubeClient := host.GetKubeClient()
+	if kubeClient == nil {
+		return fmt.Errorf("error getting kube client")
+	}
 
 	csiClient := host.GetCSIClient()
-	if csiClient != nil {
+	if (utilfeature.DefaultFeatureGate.Enabled(features.CSIDriverRegistry) ||
```

Comment:
How about only calling GetCSIClient() if either feature gate is enabled?  Then we can be sure that the csiClient will also be nil if the feature gate is not enabled.